setup/helpers/abs_helper.rb

```diff
@@ -655,23 +663,123 @@ def valid_abs_resource_hosts?(abs_resource_hosts)
     else
 
       begin
-        hosts = JSON.parse(abs_resource_hosts)
-        host = hosts[0]
-        hostname = host["hostname"]
+        hosts = parse_abs_resource_hosts(abs_resource_hosts)
+        hostname = hosts[0]["hostname"]
         if !hostname.nil? && !hostname.empty?
           is_valid = true
         else
           puts "The specified resource host array is not valid: #{abs_resource_hosts}"
           puts
         end
-      rescue StandardError
+      rescue StandardError => e
         # TODO: raise?
-        puts "JSON::ParserError encountered parsing the hosts array: #{abs_resource_hosts}"
+        puts "#{e.inspect}  encountered parsing the hosts array: #{abs_resource_hosts}"
       end
 
     end
 
     is_valid
   end
-  # rubocop:enable Naming/AccessorMethodName
+
+  # Parses the specified abs_resource_hosts JSON
+  # note: consolidating this here since it is used by multiple methods
+  #
+  # @author Bill Claytor
+  #
+  # @param [JSON] abs_resource_hosts The hosts JSON to parse
+  #
+  # @raise [RuntimeError] If parsing is not successful
+  #
+  # @return [Array] The hosts array
+  #
+  # @example
+  #   hosts = parse_abs_resource_hosts(abs_resource_hosts)
+  #
+  def parse_abs_resource_hosts(abs_resource_hosts)
+    begin
+      hosts = JSON.parse(abs_resource_hosts)
+
+    # TODO: JSON::ParserError?
+    rescue StandardError => e
+      raise "#{e.inspect} encountered parsing the hosts array: #{abs_resource_hosts}"
+    end
+    hosts
+  end
+
+  # Provisions a host with a tag for the specified role
+  # with the optionally specified size and volume_size
+  #
+  # @author Bill Claytor
+  #
+  # @param [String] role The value to specify for the role tag
+  # @param [String] size The EC2 instance size to use
+  # @param [String] volume_size The volume size to use
+  #
+  # @raise [RuntimeError] If provisioning is not successful
+  #
+  # @return [String] The hostname of the provisioned host
+  #
+  # @example
+  #   hostname = provision_host_for_role(role)
+  #     or
+  #   hostname = provision_host_for_role(role, "c5.xlarge", "40")
+  #
+  # TODO: spec test(s)
+  def provision_host_for_role(role, size = AWS_SIZE, volume_size = AWS_VOLUME_SIZE)
```

Comment:
This method should be removed.  The functionality is covered by `provision_hosts_for_roles` below.

Reply:
`provision_host_for_role` is called for each iteration of `provision_hosts_for_roles` and provides the hostname.

Reply:
Ok.